### PR TITLE
Microwave cleanup: jesus fucking christ

### PR DESCRIPTION
**Pull request by Iamgoofball**
TODO:

[] figure out the react bug (>>>reagent code)
[] refactor the new, this shit is awful
[] remove all the hacky hardcoded shit
[] type loops -> typeless
[] replace spawns with addtimers
[] readability pass
[X] oh my god it uses COLONS, exterminatus this shit
[X] let AIs use the microwave too because why only borgs the AI is a fucking AI

this file gave me Rich Evans tier AIDS

### CODE REVIEW

**Comment by Wutnold124** [comment]
goof is cleaning shit up for us

hi goof

File: code/game/machinery/kitchen/microwave.dm [hunk 1]
```diff
@@ -238,7 +238,7 @@
 	if(stat & (NOPOWER|BROKEN))
 		return
 	start()
-	if (reagents.total_volume==0 && !(locate(/obj) in ((contents - component_parts) - circuit))) //dry run
+	if (reagents.total_volume==0 && !(locate(/obj) in ((contents - component_parts) - circuit))) //dry run // >>>USING LOCATE
 		if (!wzhzhzh(10))
 			abort()
 			return
```
Comment (Iamgoofball) [inline code/game/machinery/kitchen/microwave.dm]: Just look at this fucking proc and C R Y

Comment (Wutnold124) [reply code/game/machinery/kitchen/microwave.dm]: i don't understand code but i'm going to C R Y anyways

File: code/game/machinery/kitchen/microwave.dm [hunk 2]
```diff
@@ -2,24 +2,26 @@
...
+	var/operating = 0 // Is it on? gee thanks i needed that clarified
 	var/dirty = 0 // = {0..100} Does it need cleaning?
 	var/broken = 0 // ={0,1,2} How broken is it???
-	var/global/list/datum/recipe/available_recipes // List of the recipes you can use
+	// typecache and global these
+	var/global/list/datum/recipe/available_recipes // List of the recipes you can use 
 	var/global/list/acceptable_items // List of the items you can put in
 	var/global/list/acceptable_reagents // List of the reagents you can put in
-	var/global/max_n_of_items = 0
+	var/global/max_n_of_items = 0 // why the fuck is this a global
```
Comment (Zuhayr) [inline code/game/machinery/kitchen/microwave.dm]: `global` means that all objects will share this value. Using it here means you aren't declaring identical should-be-const lists for every individual microwave.

File: code/game/machinery/kitchen/microwave.dm [hunk 3]
```diff
@@ -2,24 +2,26 @@
...
 
 
 // see code/modules/food/recipes_microwave.dm for recipes
+// see code/modules/hell/occupants.dm for coder
```
Comment (Zuhayr) [inline code/game/machinery/kitchen/microwave.dm]: This whole pull is like 90% pointless rant comments.

File: code/game/machinery/kitchen/microwave.dm [hunk 4]
```diff
@@ -51,7 +53,7 @@
 		// This will do until I can think of a fun recipe to use dionaea in -
 		// will also allow anything using the holder item to be microwaved into
 		// impure carbon. ~Z
-		acceptable_items |= /obj/item/weapon/holder
+		acceptable_items |= /obj/item/weapon/holder // I Can't Believe It's Hacky(tm)
```
Comment (Zuhayr) [inline code/game/machinery/kitchen/microwave.dm]: Go ahead and tell me how this is any hackier than any other method of making holders microwaveable.

**Comment by Anewbe** [comment]
Keep the internal comments informative, rather than derisive, please.

File: code/game/machinery/kitchen/microwave.dm (hunk 4 above)
Comment (Iamgoofball) [reply code/game/machinery/kitchen/microwave.dm]: more of a complaint about microwave code in general but i have a Feeling(TM) there's a better way to do this

**Comment by Iamgoofball** [comment]
we're all adults we can handle swear words

@Zuhayr i'm just marking them for a future fix pass in this PR

File: code/game/machinery/kitchen/microwave.dm (hunk 2 above)
Comment (Iamgoofball) [reply code/game/machinery/kitchen/microwave.dm]: dislike this on the fact that this isn't good gameplay, what if I wanted to add upgraded microwave capacity via the stock parts system(which I intend to do)

File: code/game/machinery/kitchen/microwave.dm (hunk 3 above)
Comment (Iamgoofball) [reply code/game/machinery/kitchen/microwave.dm]: a good rant helps mark shitcode, which 90% of this file is

**Comment by Anewbe** [comment]
I'm not concerned about profanity. I'm concerned about keeping the internal comments helpful, rather than derisive. Explain the shit or just leave it be, please.

**Comment by Iamgoofball** [comment]
>i'm just marking them for a future fix pass in this PR

**Comment by Fox-McCloud** [comment]
`wzhzhzh`

🤔

Also so many gosh damn implied `src`.

**Comment by Iamgoofball** [comment]
@Fox-McCloud rebase to tg when

**Comment by Iamgoofball** [comment]
Will finish this weekend

**Comment by Anewbe** [comment]
It's been five days since last comment, and I've not seen you around in that time. Closing this.

**Pull request closed without merging**